api/proto/v1/network.proto

```diff
@@ -1,3 +1,73 @@
 syntax="proto3";
 package merak;
 option go_package = "merak/";
+import "common.proto";
+
+message InternalSecurityGroupRulelnfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    string description = 4;
+    string ethertype = 5;
+    string direction = 6;
+    string protocol = 7;
+    string port_range = 8;
+    string remote_group_id = 9;
+    string remote_ip_prefix = 10;
+}
+
+message InternalRouterInfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    repeated string subnets = 4;
+}
+
+message InternalGatewayInfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    repeated string ips = 4;
+}
+
+message InternalSecurityGroupInfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    repeated string ips = 4;
+    repeated InternalSecurityGroupRulelnfo rules = 5;
+    repeated string apply_to = 6;
+}
+
+message InternalNetworkInfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    uint32 number_of_vpcs = 4;
+    uint32 number_of_subnet_per_vpc = 5;
+    repeated string subnet_ciders = 6;
+    uint32 number_of_security_groups = 7;
+    repeated InternalRouterInfo routers = 8;
+    repeated InternalGatewayInfo gateways = 9;
+    repeated InternalSecurityGroupInfo security_groups = 10;
+}
+
+message InternalServiceInfo {
+    OperationType operation_type = 1;
+    string id = 2;
+    string name = 3;
+    string cmd = 4;
+    string url = 5;
+    repeated string parameters = 6;
+    repeated string return_code = 7;
+    repeated string return_string = 8;
+    string when_to_run = 9;
+    string where_to_run = 10;
+}
+
+service MerakNetworkService {
+    rpc NetworkInfoHandler (InternalNetworkInfo) returns (InternalNetworkInfo) {}
+    rpc NetworkInfoTestHandler (InternalNetworkInfo) returns (InternalNetworkInfo) {}
+    rpc InternalServiceHandler (InternalServiceInfo) returns (InternalServiceInfo) {}
+    rpc InternalServiceTestHandler (InternalServiceInfo) returns (InternalServiceInfo) {}
```

Comment:
@yanmo96 what you plan to do for these 4 functions?

Reply:
I think those are the service to receive the messages from lines above.

Reply:
@yanmo96 Then it should only two, why there are 4?

Reply:
@yanmo96 actually, you only need one `NetConfInfoHandler`.

Reply:
The other two are the testHandler I learned from Phu's code. And as for InternalService and NetworkInfo they come in together? I will go add a new `message` to combine them then 👍

Reply:
Hi @cj-chung I have updated the code, please take a look, thanks